core/services/log/broadcaster.go

```diff
@@ -307,22 +307,24 @@ func (b *broadcaster) onNewHeads() {
 	var latestHead *models.Head
 	for {
 		// We only care about the most recent head
-		x := b.newHeads.RetrieveLatestAndClear()
-		if x == nil {
-			// This should never happen
+		item := b.newHeads.RetrieveLatestAndClear()
+		if item == nil {
```

Comment:
Could use a comment indicating how this can happen and why it's safe to ignore

Reply:
good point. Actually this always happens (ends the loop) so i'll add the comment on the `if latestHead != nil {`